gadget.core.Dockerfile

```diff
@@ -0,0 +1,100 @@
+ARG BUILDER_IMAGE=ubuntu:20.04
+ARG BASE_IMAGE=alpine:3.14
+
+# Prepare and build gadget artifacts in a container
+FROM ${BUILDER_IMAGE} as builder
+
+ARG ENABLE_BTFGEN=false
+ENV ENABLE_BTFGEN=${ENABLE_BTFGEN}
+
+RUN set -ex; \
+	export DEBIAN_FRONTEND=noninteractive; \
+	apt-get update && \
+	apt-get install -y gcc make ca-certificates git clang \
+		software-properties-common libseccomp-dev && \
+	add-apt-repository -y ppa:tuxinvader/kernel-build-tools && \
+	apt-add-repository -y ppa:longsleep/golang-backports && \
+	apt-get update && \
+	apt-get install -y libbpf-dev golang-1.17 && \
+	ln -s /usr/lib/go-1.17/bin/go /bin/go
+
+# Download BTFHub files
+COPY ./tools /btf-tools
+RUN set -ex; mkdir -p /tmp/btfs && \
+	if [ "$ENABLE_BTFGEN" = true ]; then \
+		cd /btf-tools && \
+		./getbtfhub.sh; \
+	fi
+
+# Cache go modules so they won't be downloaded at each build
+COPY go.mod go.sum /gadget/
+RUN cd /gadget && go mod download
+
+# This COPY is limited by .dockerignore
+COPY ./ /gadget
+RUN cd /gadget/gadget-container && make gadget-container-deps
+
+# Execute BTFGen
+RUN set -ex; \
+	if [ "$ENABLE_BTFGEN" = true ]; then \
+		cd /btf-tools && \
+		BTFHUB=/tmp/btfhub INSPEKTOR_GADGET=/gadget ./btfgen.sh; \
+	fi
+
+# Builder: traceloop
+
+# traceloop built from:
+# https://github.com/kinvolk/traceloop/commit/95857527df8d343a054d3754dc3b77c7c8c274c7
+# See:
+# - https://github.com/kinvolk/traceloop/actions
+# - https://hub.docker.com/r/kinvolk/traceloop/tags
+
+FROM docker.io/kinvolk/traceloop:20211109004128958575 as traceloop
+
+# Main gadget image
+FROM ${BASE_IMAGE}
+
+# install libseccomp according to the package manager available on the base image
+RUN set -ex; \
+	if command -v tdnf; then \
+		tdnf install -y libseccomp wget curl; \
+	elif command -v yum; then \
+		yum install -y libseccomp wget curl; \
+	elif command -v apt-get; then \
+		apt-get update && \
+		apt-get install -y seccompwget curl ; \
+	elif command -v apk; then \
+		apk add gcompat libseccomp bash wget curl ; \
```

Comment:
Do we forcefully need `bash` there?
I think adding bash to an alpine image increase the image size and so we loose the benefit of using alpine.

Reply:
Yes, we need bash there because our entrypoint is a bash script. Anyway, I don't think adding bash will affect the image size that much.

Reply:
> affect the image size that much.

What is our image size with alpine?
Because `bash` seems to add around 4.5 MB:

```
alpine-bash                             latest                                                   dea7fd1962e5   4 minutes ago   10MB
alpine                                  latest                                                   0ac33e5f5afa   2 weeks ago     5.57MB
```